core/tests/test-dependencies/route-to-module-mapping-generator.ts

```diff
@@ -0,0 +1,403 @@
+      resolvedModule,
+      routePath,
+      parentRoutingModuleFilePath || resolvedModule
+    );
+    routeToModuleMapping = extendMap(
+      routeToModuleMapping,
+      childRouteToModuleMapping
+    );
+  }
+  return routeToModuleMapping;
+};
+
+/**
+ * Gets the route AST nodes from a routing module file.
+ */
+const getRouteNodesFromRoutingModuleSourceFile = (
+  routingModuleSourceFile: SourceFile
+): ObjectLiteralExpression[] | undefined => {
+  const angularModuleDecorationNode = getDecorationNodesByTextFromSourceFile(
+    routingModuleSourceFile,
+    AngularDecorators.Module
+  );
+  if (angularModuleDecorationNode.length === 0) {
+    throw new Error(
+      `No Angular Module decoration was found in the file: ${routingModuleSourceFile.getFilePath()}.`
+    );
+  }
+  const angularModuleObjectArgument =
+    angularModuleDecorationNode[0].getArguments()[0];
+  if (
+    !angularModuleObjectArgument.isKind(ts.SyntaxKind.ObjectLiteralExpression)
+  ) {
+    throw new Error(
+      `The Angular Module decoration in the file: ${routingModuleSourceFile.getFilePath()} does not have an object argument.`
+    );
+  }
+  const importsProperty =
+    angularModuleObjectArgument.getPropertyOrThrow('imports');
+  const importsArray = importsProperty.getFirstChildByKindOrThrow(
+    ts.SyntaxKind.ArrayLiteralExpression
+  );
+  const routerModuleCallExpression = importsArray
+    .getElements()
+    .filter(element => {
+      return (
+        element.getText().includes('RouterModule.forRoot') ||
+        (element.getText().includes('RouterModule.forChild') &&
+          element.getKind() === ts.SyntaxKind.CallExpression)
+      );
+    })[0] as CallExpression;
+  if (routerModuleCallExpression === undefined) {
+    return undefined;
+  }
+  const routesArgument = routerModuleCallExpression.getArguments()[0];
+  if (routesArgument.isKind(ts.SyntaxKind.ArrayLiteralExpression)) {
+    return routesArgument.getElements() as ObjectLiteralExpression[];
+  }
+  if (routesArgument.isKind(ts.SyntaxKind.Identifier)) {
+    const routesVariable =
+      routingModuleSourceFile.getVariableDeclarationOrThrow(
+        routesArgument.getText()
+      );
+    return routesVariable
+      .getInitializerIfKindOrThrow(ts.SyntaxKind.ArrayLiteralExpression)
+      .getElements() as ObjectLiteralExpression[];
+  }
+  return undefined;
+};
+
+/**
+ * Gets the Angular route to module mapping from a routing module file.
+ */
+const getRouteToModuleMappingFromRoutingModule = (
+  routingModuleFilePath: string,
+  parentRoutePath?: string,
+  parentRoutingModuleFilePath?: string
+): Map<Route, string> => {
+  let routeToModuleMapping: Map<Route, string> = new Map();
+  const routingModuleSourceFile = project.addSourceFileAtPath(
+    routingModuleFilePath
+  );
+  const routingModuleRouteNodes = getRouteNodesFromRoutingModuleSourceFile(
+    routingModuleSourceFile
+  );
+  if (routingModuleRouteNodes === undefined) {
+    return routeToModuleMapping;
+  }
```

Comment:
Should this be logged?

Reply:
Hmm, here this function is used recursively on child modules, so basically it stops recursion and returns an empty map when it comes across a child module with no routes. Should it be logged in this case?

Reply:
No, lets keep it as is.